.github/workflows/main.yml

```diff
@@ -0,0 +1,35 @@
+name: "Build Docs & Push Github Pages"
+
+on: 
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

Comment:
If I understand this correctly, these two lines should be removed before merging otherwise any PR to master would update the docs.

The way tokens work though I believe this is true only for PRs opened where the author has commit rights to the repo.

Reply:
good point... I m just discovering github actions, but maybe we want 2 workflows:
- one for pull request that just attempt a build (doctests and other can be added later on)
- one for push that build and deploy

Reply:
Right now the way it works is that any change to master rebuilds the Alloy docs on readthedocs.